blog/content/post/2020-12-14-docker-testing.md

```diff
@@ -0,0 +1,375 @@
+necessary information about the version of the programs that our code uses.
+Thanks to this, everyone will be able to run it regardless of whether they use
+python 2 or 3 on their computer. Dockerfile might look like this:
+
+`Dockerfile`
+```
+FROM python:3.7-slim-stretch
+
+RUN  pip install robotframework==3.2.2 && \
+     pip install robotframework-sshlibrary==3.5.1 
+
+ENTRYPOINT ["robot"]
+```
+
+ - the first line specifies the parent image from which we are building our
+   image. More python image names can be found
+   [here](https://hub.docker.com/_/python)
+ - `RUN` runs given commands in a shell inside the container
+ - `ENTRYPOINT` specifies a command that will always be executed when the
+   container starts 
+
+In the same directory where Dockerfile is located, let's enter: 
+
+```
+mateusz@mateusz:~/blog_rf$ docker build -t blog/docker:latest .
+```
+
+When we type `docker images` we should see our image:
+
+```
+mateusz@mateusz:~/blog_rf$ docker images
+REPOSITORY               TAG                 IMAGE ID            CREATED             SIZE
+blog/docker              latest              61da82a7b715        35 minutes ago      129MB
+python                   3.7-slim-stretch    b72f35c13f06        3 weeks ago         96.9MB
+```
+
+Now we can run the tests again but this time from the container.
+
+```
+mateusz@mateusz:~/blog_rf$ docker run --rm -it -v ${PWD}:${PWD} -w ${PWD} blog/docker arduino.robot
+==============================================================================
+Arduino                                                                       
+==============================================================================
+ARD1.0 Serial Connection                                              | PASS |
+------------------------------------------------------------------------------
+ARD1.1 Pin 4 on                                                       | PASS |
+------------------------------------------------------------------------------
+ARD1.2 Pin 4 off                                                      | PASS |
+------------------------------------------------------------------------------
+ARD1.3 Pin 8 on                                                       | PASS |
+------------------------------------------------------------------------------
+ARD1.4 Pin 8 off                                                      | PASS |
+------------------------------------------------------------------------------
+Arduino                                                               | PASS |
+5 critical tests, 5 passed, 0 failed
+5 tests total, 5 passed, 0 failed
+==============================================================================
+Output:  /home/mateusz/blog_rf/output.xml
+Log:     /home/mateusz/blog_rf/log.html
+Report:  /home/mateusz/blog_rf/report.html
+```
+
+As we can see, the test results are the same except that now anyone can run our
+tests.
+
```

Comment:
Thanks for getting here, If you have found Docker handy and worthy to use, check [this source](https://containers.3mdeb.com/) of docker containers maintained by our team or forked from Open Source projects with additional useful adjustments.